Throttle expo redesign (and some cleanup)

erikmansson opened the pull request:
After testing normal expo in the transmitter for the throttle I quickly realized that it would be great to be able to do this in Betaflight instead. Rewrote the expo functions, especially for throttle. The behaviour for r/p/y stays the same. You can now select the expo style for the throttle. "FLATSPOT" is about the same as the old style, "TX_STYLE" is the same as for r/p/y (transmitter style expo). thr_mid is not used for "TX_STYLE". Also, this gets rid of the lookup table and therefore should increase the accuracy.

Sure, this is maybe a bit slower but I think it's worth it. Maybe I could add a "LINEAR" or "NO_EXPO" style which just passes though the input. That's what most of race pilots use anyway.

This PR also cleans up some whitespace.

Note that this PR requires a configurator update.

erikmansson commented:
Rebased

erikmansson commented:
Whoops

borisbstyle commented:
As the normal expo was also changed to real time it was in my planning to
do the same for throttle once, but I dont know anykne who uses it anyway so
never got the priority.

The best thing would be to be able to customize the curve through the gui
with 5 points or something like that.
Op 2 jul. 2016 17:43 schreef "Erik Månsson" <notifications@github.com>:

> Reopened #654 <https://github.com/betaflight/betaflight/pull/654>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/betaflight/betaflight/pull/654#event-711216161>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AKQlhB2sk5ywK0Ww5q6DnZ-dL4Dle5ahks5qRocfgaJpZM4JDrPF>
> .
>

erikmansson commented:
I dunno, noone I know uses 5 points... it's a bit harder to set up properly. TX style is simple and is what most of us need. I tested it flying and it works great!

borisbstyle commented:
I meant to say customisable 5 points not 5 points calculated from expo.

But i dont like the name "tx style expo". every expo has its name :)
Op 2 jul. 2016 18:16 schreef "Erik Månsson" <notifications@github.com>:

> I dunno, noone I know uses 5 points... it's a bit harder to set up
> properly. TX style is simple and is what most of us need. I tested it
> flying and it works great!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/betaflight/betaflight/pull/654#issuecomment-230109606>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKQlhJWWbv00BEv0kyzNmdD-kOOaqqk-ks5qRo7kgaJpZM4JDrPF>
> .
>

erikmansson commented:
Yeah I got that. Coming from the rc car world, I know that almost no one uses this, and almost everyone is using "tx style" for throttle/steering. But sure, there would be nothing wrong with adding that alternative!

What do you suppose the name should be? I agree, "TX_STYLE" isn't the best name. "CUBIC", "RPY", "NORMAL"?

erikmansson commented:
Tried it again for a couple of lipos now, feels great

borisbstyle commented:
I will have a look soon, but my first impression is that it might swamp some F1 processors :)

erikmansson commented:
If that's the case, I think it's a good idea to add an option to disable expo and make a simple mapping for that case. That would probably be just as fast as the old functions.

rcLookup should be just as fast, I just split it up to make it easier to read.

If you're testing flying with it, I suggest "TX_STYLE" with 20-25% expo.

erikmansson commented:
I went ahead and added a "NO_EXPO" alternative and made it as fast as possible. Renamed "TX_STYLE" to "RC". Made "NO_EXPO" default to not swamp F1 boards by default :)

Yet untested...

erikmansson commented:
Tested now, works as expected

erikmansson commented:
May I ask why you closed it?
